fiboa_cli/converter_rest.py

```diff
@@ -0,0 +1,85 @@
+from urllib.parse import urlencode
+
+import requests
+
+from .util import get_fs
+
+import os
+import geopandas as gpd
+import pandas as pd
+
+
+class EsriRESTConverterMixin:
+    cache_folder = None
+    rest_base_url = None
+    rest_params = {}
+    rest_attribute = "OBJECTID"  # orderable, filterable, indexed
+
+    def rest_layer_filter(self, layers):
+        return next(iter(layers))
+
+    def get_urls(self, **kwargs):
+        assert self.rest_base_url, "Either define {c}.rest_base_url or override {c}.get_urls()".format(c=self.__class__.__name__)
+        return {"REST": self.rest_base_url}
+
+    def download_files(self, uris, cache_folder=None):
+        # Read-data will just stream alle pages of rest-service
+        if next(iter(uris), "").startswith("REST"):
+            self.cache_folder = cache_folder
+            return list(uris.values())
+
+        # This happens when input_file param is used
+        return super().download_files(uris, cache_folder)
+
+    def read_data(self, paths, **kwargs):
+        if not paths[0].startswith("http"):
+            # This happens when input_file param is used
+            return super().read_data(paths, **kwargs)
+
+        base_url = paths[0]  # loop over paths to support more than 1 source
+
+        source_fs = get_fs(base_url)
+        cache_fs = None
+
+        if self.cache_folder:
+            cache_fs = get_fs(self.cache_folder)
+            if not cache_fs.exists(self.cache_folder):
+                cache_fs.makedirs(self.cache_folder)
```

Comment:
I think this code is copied from the convert utils. Can we make this a function so that we deduplicate the code a bit?